tests/test_batch.py

```diff
@@ -1,71 +1,91 @@
+        self.requests_mock = kwargs["mock"]
+        self.sdk = SDK(client_id="my_client_id", client_secret="my_client_secret")
+        self.pulser_sequence = "pulser_test_sequence"
+        self.batch_id = 1
+        self.job_result = "result"
+        self.n_job_runs = 50
 
-    @patch("sdk.batch.Client")
-    def test_batch_add_job(self, mock_client):
-        batch_id = 2
-        mock_client._send_job.return_value = {
-            "runs": 1,
-            "batch_id": batch_id,
-            "id": 1,
-            "status": "",
-            "created_at": "",
-            "updated_at": "",
-            "errors": [],
-        }
-        self.batch = Batch(
-            complete=False,
-            created_at="",
-            updated_at="",
-            device_type="",
-            group_id=1,
-            id=batch_id,
-            user_id=1,
-            priority=1,
-            status="",
-            webhook="",
-            _client=mock_client,
-            sequence_builder=self.pulser_sequence,
-        )
-        job = self.batch.add_job(runs=50, variables={"var": 2})
-        self.batch._client._send_job.assert_called_once()
-        assert job.batch_id == batch_id
+    def test_create_batch(self):
+        with self.requests_mock:
+            batch = self.sdk.create_batch(
+                serialized_sequence=self.pulser_sequence,
+            )
+            assert batch.id == self.batch_id
+            assert batch.sequence_builder == self.pulser_sequence
 
-    @patch("sdk.batch.Job")
-    @patch("sdk.batch.Client")
-    def test_batch_declare_complete(self, mock_job, mock_client):
-        self.batch = Batch(
-            complete=False,
-            created_at="",
-            updated_at="",
-            device_type="",
-            group_id=1,
-            id=1,
-            user_id=1,
-            priority=1,
-            status="",
-            webhook="",
-            _client=mock_client,
-            sequence_builder=self.pulser_sequence,
-        )
-        self.batch.declare_complete(wait=True)
-        self.batch._client._complete_batch.assert_called_once()
+    def test_batch_add_job(self):
+        with self.requests_mock:
+            batch = self.sdk.create_batch(
+                serialized_sequence=self.pulser_sequence,
+            )
+            job = batch.add_job(
+                runs=self.n_job_runs,
+                variables={"Omega_max": 14.4, "last_target": "q1", "ts": [200, 500]},
+            )
+            assert job.batch_id == self.batch_id
+            assert job.runs == self.n_job_runs
+
+    def test_batch_add_job_and_wait_for_results(self):
+        with self.requests_mock:
+            batch = self.sdk.create_batch(
+                serialized_sequence=self.pulser_sequence,
+            )
+            job = batch.add_job(
+                runs=self.n_job_runs,
+                variables={"Omega_max": 14.4, "last_target": "q1", "ts": [200, 500]},
+                wait=True,
```

Comment:
maybe we could assert that we called the GET /jobs/id route (I assume this is possible with request_mock) to retrieve the job results

Reply:
Yes, agreed. But if for some reason we added the result to the fixture by mistake, the assertion could have been verified even if we didn't properly fetch the results explicitly. Since fixtures are a bit artificial (and probably subject to errors / risk not being up to date) I think  we should not rely solely on them and testing the actual request is a good fixture-agnostic way to do that imo